orcid-persistence/src/main/java/org/orcid/persistence/dao/impl/EmailDaoImpl.java

```diff
@@ -78,13 +78,10 @@ public void updateEmail(String orcid, String email, boolean isCurrent, Visibilit
     @Override
     @Transactional
     public void updatePrimary(String orcid, String primaryEmail) {
-        Query updatePrimaryQuery = entityManager.createQuery("update EmailEntity set primary = 'true' where orcid = :orcid and email = :primaryEmail");
-        Query updateNonPrimaryQuery = entityManager.createQuery("update EmailEntity set primary = 'false' where orcid = :orcid and email != :primaryEmail");
-        for (Query query : new Query[] { updatePrimaryQuery, updateNonPrimaryQuery }) {
-            query.setParameter("orcid", orcid);
-            query.setParameter("primaryEmail", primaryEmail);
-            query.executeUpdate();
-        }
+        Query query = entityManager.createNativeQuery("UPDATE email SET is_primary= CASE email WHEN :primaryEmail THEN true ELSE false END WHERE orcid = :orcid");
```

Comment:
Good query, but what if I use `INFO@rcpeters.com`?

Reply:
Hey @rcpeters, this code is only used by the account settings page "Set Primary", so, it assumes the email it gets is exactly as the one in the database, we can add some more validations to it, but, it has been working like this for a while and there is no other place where we use it.

![email_set_primary](https://user-images.githubusercontent.com/464461/33668116-6ddd7d0a-da64-11e7-9e54-7e0dabdef328.png)